Code/Editor/NewLevelDialog.cpp

```diff
@@ -210,7 +211,26 @@ void CNewLevelDialog::OnLevelNameChange()
     UpdateData(true);
 
     // QRegExpValidator means the string will always be valid as long as it's not empty:
-    const bool valid = !m_level.isEmpty() && ValidateLevel();
+    bool valid = !m_level.isEmpty() && ValidateLevel();
+    if (valid)
+    {
+        QString strLevelPath = m_levelFolders + "/" + m_level + "/" + m_level + EditorUtils::LevelFile::GetDefaultFileExtension();
+        int levelMaxLength = (_MAX_PATH - m_levelFolders.length() - QString(EditorUtils::LevelFile::GetDefaultFileExtension()).length() - 2) / 2;
+        if (strLevelPath.length() >= _MAX_PATH)
```

Comment:
`_MAX_PATH` shouldn't be used to for restricting the maximum path to a level as it is windows specific.

I believe should be used is [AZ::IO::MaxPathLength](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/Path/Path_fwd.h#L38), which while larger than 260, is consistent on all platforms.
Furthermore the Max Path size can be adjusted on [Windows](https://docs.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=cmd) now, making this a user specific change